unblob/handlers/compression/lzma.py

```diff
@@ -47,19 +50,19 @@ def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]
         read_size = 0
         file.seek(start_offset + 1)
         dictionary_size = convert_int32(file.read(4), Endian.LITTLE)
+        uncompressed_size = convert_int64(file.read(8), Endian.LITTLE)
 
         # dictionary size is non-zero (section 1.1.2 of format definition)
         # dictionary size is a power of two  (section 1.1.2 of format definition)
-        if dictionary_size == 0 or (dictionary_size & (dictionary_size - 1)) != 0:
-            raise InvalidInputFormat
-
-        uncompressed_size = convert_int64(file.read(8), Endian.LITTLE)
-
         # uncompressed size is either unknown (0xFFFFFFFFFFFFFFFF) or
         # smaller than 256GB  (section 1.1.3 of format definition)
-        if not (
-            uncompressed_size == 0xFFFFFFFFFFFFFFFF
-            or uncompressed_size < MAX_UNCOMPRESSED_SIZE
+        if (
+            not (
+                uncompressed_size == 0xFFFFFFFFFFFFFFFF
+                or uncompressed_size < MAX_UNCOMPRESSED_SIZE
+            )
+            or dictionary_size == 0
+            or (dictionary_size & (dictionary_size - 1)) != 0
```

Comment:
I would keep this as two separate conditions since it helps reasoning about them and we can put dedicated comments on top. I agree with the move of `uncompressed_size` read, but not the split.

Reply:
Actually, flake8 forced me to do this. Otherwise, calculate_chunk() becomes too complex to be able to pass the tests.

Reply:
When flake8 reports such thing it's usually recommended to move the logic into a separated function.

```python
def is_valid_stream(dictionnary_size: int, uncompressed_size: int) -> bool:
    # dictionary size is non-zero (section 1.1.2 of format definition)
    # dictionary size is a power of two  (section 1.1.2 of format definition)
    if dictionary_size == 0 or (dictionary_size & (dictionary_size - 1)) != 0:
        return False
    # uncompressed size is either unknown (0xFFFFFFFFFFFFFFFF) or
    # smaller than 256GB  (section 1.1.3 of format definition)
    if not (
        uncompressed_size == 0xFFFFFFFFFFFFFFFF
        or uncompressed_size < MAX_UNCOMPRESSED_SIZE
    ):
        return False
    return True
```